src/cachingreader.cpp

```diff
@@ -11,6 +11,20 @@
 #include "util/math.h"
 #include "util/assert.h"
 
+namespace {
+
+// To prevent every bit of code having to guess how many samples
+// forward it makes sense to keep in memory, the hinter can provide
+// either 0 for a forward hint or -1 for a backward hint. We should
+// be calculating an appropriate number of samples to go backward as
+// some function of the latency, but for now just leave this as a
+// constant. 2048 is a pretty good number of samples because 25ms
```

Comment:
Ah, never mid. I have just noticed you have just copied the original comment. So fixing this issue is not part of this PR.